ofono/src/sms.c

```diff
@@ -1174,6 +1387,13 @@ static const GDBusMethodTable sms_manager_methods[] = {
 			GDBUS_ARGS({ "to", "s" }, { "text", "s" }),
 			GDBUS_ARGS({ "path", "o" }),
 			sms_send_message) },
+	{ GDBUS_ASYNC_METHOD("SendDataMessage",
+			GDBUS_ARGS(
+				{ "to", "s" },
+				{ "data", "v" }, // s|ay
```

Comment:
It's a data SMS, hence `ay`. Text messages are properly encoded, packed and sent by `SendMessage`.

Reply:
Well I think you misunderstood what I was aiming here. I really see no harm in having Ofono to encode the text into the requested payload content when requested. The functionality is already there, it is better to use the encoding from within ofono where it is maintained in one central place instead of having the encoding part duplicated into multiple components that use this feature.

Reply:
7-bit encoding is too simple to justify complication of the API. `SendDataMessage` will send opaque binary data, without any transformation applied. `SendTextMessage` will send text and apply all the necessary transformation.